src/main/java/no/nav/pto/veilarbportefolje/persononinfo/barnUnder18Aar/BarnUnder18AarService.java

```diff
@@ -51,37 +53,46 @@ public List<Fnr> hentBarnFnrsForForeldre(List<Fnr> fnrForeldre) {
 
             lagredeBarn.forEach(barnFnr -> {
-                if (pdlPersonBarn == null || !pdlPersonBarn.containsKey(barnFnr)) {
-                    secureLog.warn("Slett foreldre ansvar, master: " + String.join(",", foreldreansvarMaster));
+                if (foreldreansvarPDL == null || !foreldreansvarPDL.contains(barnFnr)) {
+                    secureLog.warn("Slett foreldreansvar");
                     slettForeldreansvar(foresattIdent, barnFnr);
                     slettBarnDataHvisIngenForeldreErUnderOppfolging(barnFnr);
                 }
             });
 
-            if (pdlPersonBarn != null && !pdlPersonBarn.isEmpty()){
-                for (var barnMap : pdlPersonBarn.entrySet()) {
-                    if (barnMap.getValue().isErIlive() && erUnder18Aar(barnMap.getValue().getFodselsdato())) {
-                        barnUnder18AarRepository.lagreBarnData(barnMap.getKey(), barnMap.getValue().getFodselsdato(), barnMap.getValue().getDiskresjonskode());
-                        barnUnder18AarRepository.lagreForeldreansvar(foresattIdent, barnMap.getKey());
-                    }else{
-                        slettForeldreansvar(foresattIdent, barnMap.getKey());
-                        slettBarnDataHvisIngenForeldreErUnderOppfolging(barnMap.getKey());
-                    }
-                }
+            nyeBarn = foreldreansvarPDL.stream().filter(barnFnr -> !lagredeBarn.contains(barnFnr)).toList();
+            if (!nyeBarn.isEmpty()) {
+                getDataFromPDLAndInsertBarn(foresattIdent, nyeBarn);
             }
-        }
-        catch (Exception e){
-            secureLog.error("Kan ikke lagre data om barn og foreldreansvar for person: " + foresattIdent + ". Antall barn: " + pdlPersonBarn.size() + ", error: " + e.getMessage(), e);
+        } catch (Exception e) {
+            secureLog.error("Kan ikke lagre data om barn og foreldreansvar for person: " + foresattIdent + ". Antall barn: " + nyeBarn.size() + ", error: " + e.getMessage(), e);
             throw new RuntimeException("Kan ikke lagre data om barn");
         }
     }
 
 
+    public void getDataFromPDLAndInsertBarn(Fnr foresattIdent, List<Fnr> nyeBarn) {
+        Map<Fnr, PDLPersonBarn> pdlPersonBarn = pdlClient.hentBrukerBarnDataBolkFraPdl(nyeBarn);
+        if (pdlPersonBarn != null && !pdlPersonBarn.isEmpty()) {
+            pdlPersonBarn.forEach((fnrBarn, dataBarn) -> {
+                if (dataBarn.isErIlive() && erUnder18Aar(dataBarn.getFodselsdato())) {
+                    barnUnder18AarRepository.lagreBarnData(fnrBarn, dataBarn.getFodselsdato(), dataBarn.getDiskresjonskode());
+                    barnUnder18AarRepository.lagreForeldreansvar(foresattIdent, fnrBarn);
+                } else {
+                    slettForeldreansvar(foresattIdent, fnrBarn);
+                    slettBarnDataHvisIngenForeldreErUnderOppfolging(fnrBarn);
+                }
```

Comment:
@OdaDahlen I suppose we dont need this, since this is new barn and we can't delete any foreldreansvar. What do you think? Am I missing something..